src/v/cluster/self_test_rpc_types.h

```diff
@@ -322,15 +330,17 @@ struct get_status_response
 
     uuid_t id{};
     self_test_status status{};
+    self_test_stage stage{};
```

Comment:
The addition of `stage` seems independent of the cloud check itself. Could you pull it out into a separate commit -- perhaps even a separate PR, and then base this one on it

Reply:
Pulled out all `stage` changes to separate commit.